src/inference/tests/functional/get_supported_property_test.cpp

```diff
@@ -0,0 +1,176 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include <string>
+
+#include "common_test_utils/file_utils.hpp"
+#include "cpp_interfaces/interface/ie_iplugin_internal.hpp"
+#include "ie_core.hpp"
+#include "ie_icore.hpp"
+#include "openvino/openvino.hpp"
+#include "openvino/runtime/properties.hpp"
+#include "openvino/util/file_util.hpp"
+#include "openvino/util/shared_object.hpp"
+
+namespace {
+std::string get_mock_engine_path() {
+    std::string mockEngineName("mock_engine");
+    return ov::util::make_plugin_library_name(CommonTestUtils::getExecutableDirectory(),
+                                              mockEngineName + IE_BUILD_POSTFIX);
+}
+ov::PropertyName RO_property(const std::string& propertyName) {
+    return ov::PropertyName(propertyName, ov::PropertyMutability::RO);
+}
+ov::PropertyName RW_property(const std::string& propertyName) {
+    return ov::PropertyName(propertyName, ov::PropertyMutability::RW);
+}
+template <class T>
+std::function<T> make_std_function(const std::shared_ptr<void> so, const std::string& functionName) {
+    std::function<T> ptr(reinterpret_cast<T*>(ov::util::get_symbol(so, functionName.c_str())));
+    return ptr;
+}
+
+}  // namespace
+
+class MockHardWarePlugin : public InferenceEngine::IInferencePlugin {
```

Comment:
Can we write this test using new plugin API?

Reply:
fixed to new plugin API